core/src/main/java/com/linecorp/armeria/client/endpoint/WeightRampingUpStrategy.java

```diff
@@ -144,9 +145,14 @@ final class RampingUpEndpointWeightSelector extends AbstractEndpointSelector {
         }
 
         @Override
-        protected void updateNewEndpoints(List<Endpoint> endpoints) {
+        protected CompletableFuture<Void> updateNewEndpoints(List<Endpoint> endpoints) {
             // Use the executor so the order of endpoints change is guaranteed.
-            executor.execute(() -> updateEndpoints(endpoints));
+            final CompletableFuture<Void> cf = new CompletableFuture<>();
+            executor.execute(() -> {
+                updateEndpoints(endpoints);
+                cf.complete(null);
+            });
+            return cf;
```

Comment:
`runAsync` is simpler and propagates exceptions raised in `updateEndpoints()`.
```suggestion
            CompletableFuture.runAsync(() -> updateEndpoints(endpoints), executor);
```

Reply:
On second thought, what do you think of using a lock instead of an executor for the thread safety?

Reply:
I think it's doable as long as a fair lock is used to guarantee the `endpoints` update order.